models/stock_picking.py

```diff
@@ -322,14 +324,96 @@ def update_picking(
             # mark_as_done the stock.move.lines
             mls_done = move_lines.mark_as_done(**values)
 
+            if real_time_update:
+                picking = self._real_time_update(mls_done)
+                validate = True
+
         if force_validate or validate:
-            if self.move_line_ids.get_lines_todo() and not create_backorder:
+            if picking.move_line_ids.get_lines_todo() and not create_backorder:
                 raise ValidationError(
                         _('Cannot validate transfer because there'
                           ' are move lines todo'))
             # by default action_done will backorder the stock.move.lines todo
             # validate stock.picking
-            self.action_done() # old do_transfer
+            picking.action_done() # old do_transfer
+
+    def _requires_backorder(self, mls):
+        # if we
+        #   have all the moves
+        #   have all the move_lines within them
+        #   are not waiting for any previous moves to complete
+        # then we dont need to backorder
+        mls_moves = mls.mapped('move_id')
+        for move in self.move_lines:
+            if move not in mls_moves:
+                break
+            if not move.move_line_ids == mls.filtered(lambda x: x.move_id == move):
+                break
+            if move.mapped('move_orig_ids').filtered(lambda x: x.state not in ('done', 'cancel')):
+                break
+        else:
+            return False
+        return True
+
+    def _create_backorder(self, mls):
```

Comment:
Make mls an optional parameter and when it is not set get all the stock.move.lines done of the picking?